src/python/pants/base/exception_sink.py

```diff
@@ -454,12 +437,3 @@ def _handle_signal_gracefully(cls, signum, signame, traceback_lines):
 
 # Setup a default signal handler.
 ExceptionSink.reset_signal_handler(SignalHandler(pantsd_instance=False))
-
-# Set whether to print stacktraces on exceptions or signals during import time.
-# NB: This will be overridden by bootstrap options in PantsRunner, so we avoid printing out a full
-# stacktrace when a user presses control-c during import time unless the environment variable is set
-# to explicitly request it. The exception log will have any stacktraces regardless so this should
-# not hamper debugging.
-ExceptionSink.reset_should_print_backtrace_to_terminal(
-    should_print_backtrace=os.environ.get("PANTS_PRINT_EXCEPTION_STACKTRACE", "True") == "True"
```

Comment:
If I understand correctly, do we *always* print the stacktrace if we receive an exception at import time? Or (just to make sure) does this turn *off* stacktraces for errors at import time?

Reply:
I don't think this code actually has anything to do with whether we print stacktraces on exceptions at import time. The only two places in the pants codebase that were actually paying attention to this variable were the `ExceptionFormatter` in `init/logging.py` and `_handle_signal_gracefully` in `ExceptionSink`. The former now just directly pays attention to `--print-exception-stacktrace` and the latter always prints a stack trace but only gets invoked when there's a SIGTERM or SIGQUIT signal.

Reply:
I think what's going to happen is that at import time (so, before the logging initialization code has run), if there's an exception, it will be handled the same way as exceptions are handled by default in any python program - so, an exception traceback will be printed to the terminal.

Reply:
I'm just concerned that that won't work if we have pantsd active, which was the concern for #6530. If we haven't seen it being an issue, then I think we can suspect as you say that that code has moved to `init/logging.py` at some point.